sdk/eventplugin/eventplugin.proto

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+syntax = "proto3";
+
+package eventplugin.v1;
+
+import "google/protobuf/struct.proto";
+import "sdk/plugin/pb/backend.proto";
+import "sdk/logical/event.proto";
+
+option go_package = "github.com/hashicorp/vault/sdk/eventplugin";
+
+message InitializeRequest {}
+message InitializeResponse {}
+
+message SubscriptionEvent {
+	// only a single subscription ID will be used in a stream, so the value MAY be omitted in subsequent messages,
+	// but is REQUIRED for the first message. This is included for future-proofing having a plugin handle multiple
+	// subscriptions per plugin instance.
+	string subscription_id = 1;
+	string eventJson = 2;
+}
+
+message SubscribeRequest {
```

Comment:
Is there a way to subscribe to a grouping of events, or is each one a separate thing?

Reply:
It's flexible -- this is a low-level interface that is more about plumbing events to SQS (e.g.), but it will depend on how we want to configure the endpoint that matches up event topics / bexpr to the subscription.

My current inclination is to support globs in the event topic and bexpr, similar to how we do websockets, so that the interfaces are similar.

Reply:
How will we reference a particular events plugin mount in that matching up endpoint, and where in the API are we hosting this? Will it all be in the sys/ mount? In particular, secrets engines give us a bit of a headache because they can occupy basically any path that isn't already reserved for e.g. auth/ or sys/